File: benchmarks/bench_fibonacci_iter.py
"""Iterative Fibonacci benchmark - matches bench_fibonacci.nlpl (Fibonacci(1000))."""
import time


def fibonacci(n: int) -> int:
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b


if __name__ == "__main__":
    start = time.perf_counter()
    result = fibonacci(1000)
    elapsed = time.perf_counter() - start
    print(f"Fibonacci(1000) = {result}")
    print(f"Time: {elapsed:.9f} seconds")

File: benchmarks/bench_fibonacci_iter.rs
// Iterative Fibonacci benchmark - matches bench_fibonacci.nlpl (Fibonacci(1000))
use std::time::Instant;

fn fibonacci(n: u64) -> u64 {
    if n <= 1 {
        return n;
    }
    let (mut a, mut b) = (0u64, 1u64);
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    let start = Instant::now();
    let result = fibonacci(1000);
    let elapsed = start.elapsed();
    println!("Fibonacci(1000) = {}", result);
    println!("Time: {:.9} seconds", elapsed.as_secs_f64());
}
